### DISCUSSION TITLE
StatefulSet replica set to 2

### DISCUSSION THREAD

**Discussion by saoDiSengTank** [metacontroller/metacontroller#794] [Q&A]
Hi Team ,

i am trying to set the StatefulSet replica value to 2 but got some strange results, that's why i am here to ask some questions.
1. Why we using StatefulSet , not deployment to create pods ?
2. After replica set 2 , our service received the reconcile 2 times when i deploy the crd to trigger DecoratorController , does that result correct ? If yes , is there a param to  close that ? If not , what perhaps the root cause ?
3.  Our reconcile triggered every 30mins after crd deployed , but  the reconcile interval changed to 15min if replica set 2 , is that correct ?

Thanks

**Comment by grzesuav** [comment]
hi @saoDiSengTank :
1. As by default leader election is not set, and StS and Deployment have different semantic in case of split-brain scenario
2. Yes, each of the replicas reconciles concurrently and independently, use `leader election`
3. Hmm not really, I guess it might be due to having two replicas reconciling  indenedently
